### PR TITLE
cross product of two vectors

### PR DESCRIPTION
**Pull request by St-Maxwell**
Cross product of two vectors.
#457

### CODE REVIEW

File: src/stdlib_linalg_cross_product.fypp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#:set RCI_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES + INT_KINDS_TYPES
+submodule (stdlib_linalg) stdlib_linalg_cross_product
+
+    implicit none
+
+contains
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+    pure module function cross_product_${t1[0]}$${k1}$(a, b) result(res)
+        ${t1}$, intent(in) :: a(3), b(3)
```
Comment (jvdp1) [inline src/stdlib_linalg_cross_product.fypp:11]:
```suggestion
        ${t1}$, intent(in) :: a(:), b(:)
```
Just a question: why did you choose to have `a(3)` and `b(3)`?

Comment (jvdp1) [reply src/stdlib_linalg_cross_product.fypp:11]: Great. Thank you.

File: test/linalg/test_linalg.fypp [hunk 2]
```diff
@@ -702,6 +712,163 @@ contains
...
+        integer, parameter :: n = 3
+        integer(int8) :: u(n), v(n), expected(n), diff(n)
+
+        u = [1,0,0]
+        v = [0,1,0]
+        expected = [0,0,1]
+        diff = expected - cross_product(u,v)
+        call check(error, all(abs(diff) == 0), &
+             "all(abs(diff) == 0) failed.")
+    end subroutine test_cross_product_int8
```
Comment (jvdp1) [inline test/linalg/test_linalg.fypp:728]: I suggest to use `fypp` to generate all these (similar) procedures.

File: test/linalg/test_linalg.fypp [hunk 3]
```diff
@@ -702,6 +712,163 @@ contains
...
+
+        integer, parameter :: n = 3
+        integer(int8) :: u(n), v(n), expected(n), diff(n)
+
+        u = [1,0,0]
+        v = [0,1,0]
+        expected = [0,0,1]
+        diff = expected - cross_product(u,v)
+        call check(error, all(abs(diff) == 0), &
+             "all(abs(diff) == 0) failed.")
```
Comment (jvdp1) [inline test/linalg/test_linalg.fypp:727]: This message might be more explicit

**Review by jvdp1 — COMMENTED** [review]
Thank you. LGTM. I have only a few minor comments.

**Review by jvdp1 — APPROVED** [review]

**Review by milancurcic — APPROVED** [review]
Thanks! This looks good.

**Pull request merged**